test/TestProjects/MgmtSingleton/Generated/SingletonResource.cs

```diff
@@ -60,5 +72,309 @@ public virtual SingletonResourceData Data
                 return _data;
             }
         }
+
+        /// <summary> Gets the parent resource of this resource. </summary>
+        public ArmResource Parent { get; }
+
+        /// <summary> Singleton Test Example. See /providers/Microsoft.Billing/billingAccounts/{billingAccountName}/billingProfiles/{billingProfileName}/policies/default GET,PUT. </summary>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public async virtual Task<Response<SingletonResource>> GetAsync(CancellationToken cancellationToken = default)
```

Comment:
Can we add some tests to verify that the methods are generated in the correct classes? For singleton resource, you can add some tests [here](https://github.com/Azure/autorest.csharp/blob/ba750cbf6bf1756f9b29d4f97dfeee93990a7141/test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtSingletonTests.cs) to verify this.

Reply:
Tests have been added